storage/mediator.go

```diff
@@ -199,3 +219,33 @@ func (m *mediator) reportLoop() {
 		}
 	}
 }
+
+func (m *mediator) databaseBootstrapStates() (databaseBootstrapStates, error) {
+	namespaces, err := m.database.GetOwnedNamespaces()
+	if err != nil {
+		return databaseBootstrapStates{}, err
+	}
+
+	states := databaseBootstrapStates{
+		namespaceBootstrapStates: make(map[string]shardBootstrapStates, len(namespaces)),
+	}
+
+	for _, namespace := range namespaces {
+		nsState := make(shardBootstrapStates)
+
+		for _, shard := range namespace.GetOwnedShards() {
+			nsState[shard.ID()] = shard.IsBootstrapped()
+		}
+		states.namespaceBootstrapStates[namespace.ID().String()] = nsState
+	}
+
+	return states, nil
+}
+
+// databaseBootstrapStates storse a snapshot of the bootstrap state for all shards across all
+// namespaces at a given moment in time.
+type databaseBootstrapStates struct {
+	namespaceBootstrapStates map[string]shardBootstrapStates
+}
+
+type shardBootstrapStates map[uint32]bool
```

Comment:
should this be `map[uint32]bootstrapState`? can migrate the `bootstrapState` enum decl to be along side this type. and then `shard.IsBootstrapped` can be changed to `shard.BootstrapState()`

Reply:
Makes sense, I Might keep in IsBootstrapped() thought and just have it call BootstrapState()

Reply:
done